File: 2005/Windows SDK NetFx 3.0/Applications/CultureSample/CS/CultureSample/NewInstance.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using System.Globalization;
namespace Microsoft.Samples.Globalization.Culture
{
	/// <summary>
	/// Summary description for form.
	/// </summary>
	public class NewInstance : System.Windows.Forms.Form
	{
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.IContainer components = null;
		private System.Windows.Forms.Label labelHeading;
		private System.Windows.Forms.Button buttonCancel;
		private System.Windows.Forms.Label labelName;
		private System.Windows.Forms.ComboBox comboBoxLocale;
		private System.Windows.Forms.TextBox textBoxName;
		private System.Windows.Forms.Label labelLocale;
		private System.Windows.Forms.Button buttonNext;

		//private CultureInfoHelper to be passed on to the child form
		private CultureInfoHelper helper;

		public NewInstance(CultureInfoHelper thisHelper)
		{
			InitializeComponent();
			helper = thisHelper;
			comboBoxLocale.Items.AddRange(helper.GetCultures(
				CultureTypes.InstalledWin32Cultures));
			comboBoxLocale.SelectedIndex = 0;
		}

		private void buttonCancel_Click(object sender, System.EventArgs e)
		{
			this.Close();
		}

		private void buttonNext_Click_1(object sender, System.EventArgs e)
		{
			if (textBoxName.Text.Length == 0)
				return;
			if (textBoxName.Text.Contains(Constants.Space))
			{
				MessageBox.Show(Constants.ErrorInvalidName);
				textBoxName.Clear();
				return;
			}
			//Using the helper register the new instance
			helper.NewCultureInstance((string)comboBoxLocale.SelectedItem, textBoxName.Text);
			this.Close();
		}
		
		#region Windows designer code
		private void InitializeComponent()
		{
			this.labelHeading = new System.Windows.Forms.Label();
			this.buttonCancel = new System.Windows.Forms.Button();
			this.labelName = new System.Windows.Forms.Label();
			this.comboBoxLocale = new System.Windows.Forms.ComboBox();
			this.textBoxName = new System.Windows.Forms.TextBox();
			this.labelLocale = new System.Windows.Forms.Label();
			this.buttonNext = new System.Windows.Forms.Button();
			this.SuspendLayout();
// 
// labelHeading
// 
			this.labelHeading.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
			this.labelHeading.Location = new System.Drawing.Point(84, 34);
			this.labelHeading.Name = "labelHeading";
			this.labelHeading.Size = new System.Drawing.Size(142, 23);
			this.labelHeading.TabIndex = 22;
			this.labelHeading.Text = "Create a new instance";
// 
// buttonCancel
// 
			this.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
			this.buttonCancel.Location = new System.Drawing.Point(151, 210);
			this.buttonCancel.Name = "buttonCancel";
			this.buttonCancel.TabIndex = 29;
			this.buttonCancel.Text = "Cancel";
			this.buttonCancel.Click += new System.EventHandler(this.buttonCancel_Click);
// 
// labelName
// 
			this.labelName.Location = new System.Drawing.Point(36, 140);
			this.labelName.Name = "labelName";
			this.labelName.Size = new System.Drawing.Size(63, 23);
			this.labelName.TabIndex = 23;
			this.labelName.Text = "Name:";
// 
// comboBoxLocale
// 
			this.comboBoxLocale.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
			this.comboBoxLocale.Location = new System.Drawing.Point(115, 104);
			this.comboBoxLocale.Name = "comboBoxLocale";
			this.comboBoxLocale.Size = new System.Drawing.Size(142, 21);
			this.comboBoxLocale.TabIndex = 26;
// 
// textBoxName
// 
			this.textBoxName.Location = new System.Drawing.Point(115, 140);
			this.textBoxName.Name = "textBoxName";
			this.textBoxName.Size = new System.Drawing.Size(142, 20);
			this.textBoxName.TabIndex = 27;
// 
// labelLocale
// 
			this.labelLocale.Location = new System.Drawing.Point(36, 104);
			this.labelLocale.Name = "labelLocale";
			this.labelLocale.Size = new System.Drawing.Size(57, 23);
			this.labelLocale.TabIndex = 24;
			this.labelLocale.Text = "Locale:";
// 
// buttonNext
// 
			this.buttonNext.Location = new System.Drawing.Point(52, 210);
			this.buttonNext.Name = "buttonNext";
			this.buttonNext.TabIndex = 28;
			this.buttonNext.Text = "OK";
			this.buttonNext.Click += new System.EventHandler(this.buttonNext_Click_1);
// 
// NewInstance
// 
			this.AcceptButton = this.buttonNext;
			this.AutoSize = true;
			this.CancelButton = this.buttonCancel;
			this.ClientSize = new System.Drawing.Size(292, 266);
			this.Controls.Add(this.buttonNext);
			this.Controls.Add(this.labelName);
			this.Controls.Add(this.comboBoxLocale);
			this.Controls.Add(this.textBoxName);
			this.Controls.Add(this.labelLocale);
			this.Controls.Add(this.labelHeading);
			this.Controls.Add(this.buttonCancel);
			this.Name = "NewInstance";
			this.Padding = new System.Windows.Forms.Padding(9);
			this.Text = "New Instance";
			this.ResumeLayout(false);
			this.PerformLayout();

		}
		#endregion
		
		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(disposing);
		}


	}
}

File: 2005/Windows SDK NetFx 3.0/Applications/CultureSample/VB/CultureSample/NewInstance.vb
 '-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
'Imports System.Data

Imports System.Globalization

'/ <summary>
'/ Summary description for form.
'/ </summary>

Public Class NewInstance
    Inherits System.Windows.Forms.Form
    '/ <summary>
    '/ Required designer variable.
    '/ </summary>
    Private components As System.ComponentModel.IContainer = Nothing
    Private labelHeading As System.Windows.Forms.Label
    Private WithEvents buttonCancel As System.Windows.Forms.Button
    Private labelName As System.Windows.Forms.Label
    Private comboBoxLocale As System.Windows.Forms.ComboBox
    Private textBoxName As System.Windows.Forms.TextBox
    Private labelLocale As System.Windows.Forms.Label
    Private WithEvents buttonNext As System.Windows.Forms.Button
    
    'private CultureInfoHelper to be passed on to the child form
    Private helper As CultureInfoHelper
    
    
    Public Sub New(ByVal thisHelper As CultureInfoHelper) 
        InitializeComponent()
        helper = thisHelper
        comboBoxLocale.Items.AddRange(helper.GetCultures(CultureTypes.InstalledWin32Cultures))
        comboBoxLocale.SelectedIndex = 0
    
    End Sub 'New
    
    
    Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As System.EventArgs)  Handles buttonCancel.Click
        Me.Close()
    
    End Sub 'buttonCancel_Click
    
    
    Private Sub buttonNext_Click_1(ByVal sender As Object, ByVal e As System.EventArgs)  Handles buttonNext.Click
        If textBoxName.Text.Length = 0 Then
            Return
        End If
        If textBoxName.Text.Contains(Constants.Space) Then
            MessageBox.Show(Constants.ErrorInvalidName)
            textBoxName.Clear()
            Return
        End If
        'Using the helper register the new instance
        helper.NewCultureInstance(CStr(comboBoxLocale.SelectedItem), textBoxName.Text)
        Me.Close()
    
    End Sub 'buttonNext_Click_1
    
    #Region "Windows designer code"
    
    Private Sub InitializeComponent() 
        Me.labelHeading = New System.Windows.Forms.Label()
        Me.buttonCancel = New System.Windows.Forms.Button()
        Me.labelName = New System.Windows.Forms.Label()
        Me.comboBoxLocale = New System.Windows.Forms.ComboBox()
        Me.textBoxName = New System.Windows.Forms.TextBox()
        Me.labelLocale = New System.Windows.Forms.Label()
        Me.buttonNext = New System.Windows.Forms.Button()
        Me.SuspendLayout()
        ' 
        ' labelHeading
        ' 
        Me.labelHeading.Font = New System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, System.Convert.ToByte(0))
        Me.labelHeading.Location = New System.Drawing.Point(84, 34)
        Me.labelHeading.Name = "labelHeading"
        Me.labelHeading.Size = New System.Drawing.Size(142, 23)
        Me.labelHeading.TabIndex = 22
        Me.labelHeading.Text = "Create a new instance"
        ' 
        ' buttonCancel
        ' 
        Me.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.buttonCancel.Location = New System.Drawing.Point(151, 210)
        Me.buttonCancel.Name = "buttonCancel"
        Me.buttonCancel.TabIndex = 29
        Me.buttonCancel.Text = "Cancel"
        ' 
        ' labelName
        ' 
        Me.labelName.Location = New System.Drawing.Point(36, 140)
        Me.labelName.Name = "labelName"
        Me.labelName.Size = New System.Drawing.Size(63, 23)
        Me.labelName.TabIndex = 23
        Me.labelName.Text = "Name:"
        ' 
        ' comboBoxLocale
        ' 
        Me.comboBoxLocale.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
        Me.comboBoxLocale.Location = New System.Drawing.Point(115, 104)
        Me.comboBoxLocale.Name = "comboBoxLocale"
        Me.comboBoxLocale.Size = New System.Drawing.Size(142, 21)
        Me.comboBoxLocale.TabIndex = 26
        ' 
        ' textBoxName
        ' 
        Me.textBoxName.Location = New System.Drawing.Point(115, 140)
        Me.textBoxName.Name = "textBoxName"
        Me.textBoxName.Size = New System.Drawing.Size(142, 20)
        Me.textBoxName.TabIndex = 27
        ' 
        ' labelLocale
        ' 
        Me.labelLocale.Location = New System.Drawing.Point(36, 104)
        Me.labelLocale.Name = "labelLocale"
        Me.labelLocale.Size = New System.Drawing.Size(57, 23)
        Me.labelLocale.TabIndex = 24
        Me.labelLocale.Text = "Locale:"
        ' 
        ' buttonNext
        ' 
        Me.buttonNext.Location = New System.Drawing.Point(52, 210)
        Me.buttonNext.Name = "buttonNext"
        Me.buttonNext.TabIndex = 28
        Me.buttonNext.Text = "OK"
        ' 
        ' NewInstance
        ' 
        Me.AcceptButton = Me.buttonNext
        Me.AutoSize = True
        Me.CancelButton = Me.buttonCancel
        Me.ClientSize = New System.Drawing.Size(292, 266)
        Me.Controls.Add(buttonNext)
        Me.Controls.Add(labelName)
        Me.Controls.Add(comboBoxLocale)
        Me.Controls.Add(textBoxName)
        Me.Controls.Add(labelLocale)
        Me.Controls.Add(labelHeading)
        Me.Controls.Add(buttonCancel)
        Me.Name = "NewInstance"
        Me.Padding = New System.Windows.Forms.Padding(9)
        Me.Text = "New Instance"
        Me.ResumeLayout(False)
        Me.PerformLayout()
    
    End Sub 'InitializeComponent 
    #End Region
    
    
    '/ <summary>
    '/ Clean up any resources being used.
    '/ </summary>
    Protected Overrides Sub Dispose(ByVal disposing As Boolean) 
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    
    End Sub 'Dispose
End Class 'NewInstance 
